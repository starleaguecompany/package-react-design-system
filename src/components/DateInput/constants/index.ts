export const DAYS_OF_WEEK = ['Пн', 'Вт', 'Ср', 'Чт', 'Пт', 'Сб', 'Вс']
export const MONTHS = [
  'Январь',
  'Февраль',
  'Март',
  'Апрель',
  'Май',
  'Июнь',
  'Июль',
  'Август',
  'Сентябрь',
  'Октябрь',
  'Ноябрь',
  'Декабрь',
]
