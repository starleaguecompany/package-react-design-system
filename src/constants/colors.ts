export const COLORS = {
  GRAY: 'gray',
  GREEN: 'green',
  BLUE: 'blue',
  ORANGE: 'orange',
  VIOLET: 'violet',
  RED: 'red',
} as const
