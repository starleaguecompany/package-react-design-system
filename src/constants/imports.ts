export const reactDSImportPath = '@starleaguecompany/react-design-system';
