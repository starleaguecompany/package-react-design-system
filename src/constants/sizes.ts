export const CONTENT_SIZES = {
  S12: 12,
  S16: 16,
  S20: 20,
  S24: 24,
  S28: 28,
} as const;

export const CONTAINER_SIZES = {
  S36: 36,
  S44: 44,
  S52: 52,
  S60: 60,
} as const;

export const FONT_SIZES = {
  S10: 10,
  S12: 12,
  S14: 14,
  S16: 16,
  S20: 20,
} as const;
