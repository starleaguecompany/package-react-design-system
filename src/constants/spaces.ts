export const SPACE_SIZES = {
  S0: 0,
  S2: 2,
  S4: 4,
  S8: 8,
  S12: 12,
  S16: 16,
  S20: 20,
  S24: 24,
  S28: 28,
  S32: 32,
  S36: 36,
  S40: 40,
} as const;
