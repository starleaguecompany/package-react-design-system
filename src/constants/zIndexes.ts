export const Z_INDEXES = {
  DROPDOWN: 1060,
  STICKY: 1020,
  FIXED: 1030,
  MODAL_BACKDROP: 1040,
  MODAL: 1050,
  TOOLTIP: 1060,
} as const;
