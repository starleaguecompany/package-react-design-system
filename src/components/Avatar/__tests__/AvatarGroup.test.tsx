import * as React from 'react';
import { render, screen } from '@testing-library/react';

import { CONTAINER_SIZES } from '../../../constants/sizes';
import { Avatar, AvatarGroup } from '../index';

describe('AvatarGroup', () => {
  test('should render correctly with no props', () => {
    render(<AvatarGroup />);

    expect(screen.getByTestId('AvatarGroup')).toMatchSnapshot();
    expect(screen.getByTestId('AvatarGroup')).toHaveClass('group', { exact: true });
  });

  test('should render correctly with attributes', () => {
    render(<AvatarGroup id="test-id" className="test-class" />);

    expect(screen.getByTestId('AvatarGroup')).toMatchSnapshot();
    expect(screen.getByTestId('AvatarGroup')).toHaveAttribute('id', 'test-id');
    expect(screen.getByTestId('AvatarGroup')).toHaveClass('test-class');
  });

  test('should render correctly with avatars', () => {
    render(
      <AvatarGroup>
        <Avatar src="test" />
        <Avatar src="test" />
      </AvatarGroup>
    );

    expect(screen.getByTestId('AvatarGroup')).toMatchSnapshot();
    expect(screen.getAllByRole('img')[0]).toBeVisible();
    expect(screen.getAllByRole('img')[1]).toBeVisible();
    expect(screen.getAllByTestId('Avatar').length).toEqual(2);
  });

  test('should render correctly with size prop', () => {
    render(
      <AvatarGroup size={CONTAINER_SIZES.S60}>
        <Avatar src="test" />
        <Avatar src="test" />
      </AvatarGroup>
    );

    expect(screen.getByTestId('AvatarGroup')).toMatchSnapshot();
    expect(screen.getAllByTestId('Avatar')[0]).toHaveClass('size-60');
    expect(screen.getAllByTestId('Avatar')[1]).toHaveClass('size-60');
  });
});
