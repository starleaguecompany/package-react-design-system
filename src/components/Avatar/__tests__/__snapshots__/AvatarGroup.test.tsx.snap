// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AvatarGroup should render correctly with attributes 1`] = `
<div
  class="test-class group"
  data-qa="AvatarGroup"
  id="test-id"
/>
`;

exports[`AvatarGroup should render correctly with avatars 1`] = `
<div
  class="group"
  data-qa="AvatarGroup"
>
  <div
    class="avatar size-36"
    data-qa="Avatar"
  >
    <img
      alt="avatar"
      class="image"
      loading="lazy"
      src="test"
    />
  </div>
  <div
    class="avatar size-36"
    data-qa="Avatar"
  >
    <img
      alt="avatar"
      class="image"
      loading="lazy"
      src="test"
    />
  </div>
</div>
`;

exports[`AvatarGroup should render correctly with no props 1`] = `
<div
  class="group"
  data-qa="AvatarGroup"
/>
`;

exports[`AvatarGroup should render correctly with size prop 1`] = `
<div
  class="group"
  data-qa="AvatarGroup"
>
  <div
    class="avatar size-60"
    data-qa="Avatar"
  >
    <img
      alt="avatar"
      class="image"
      loading="lazy"
      src="test"
    />
  </div>
  <div
    class="avatar size-60"
    data-qa="Avatar"
  >
    <img
      alt="avatar"
      class="image"
      loading="lazy"
      src="test"
    />
  </div>
</div>
`;
